lib/services/service_brokers/service_manager.rb

```diff
@@ -58,16 +59,37 @@ def update_or_create_plans(catalog)
           plan.public = false
         end
 
+        create_instance = nil
+        begin
+          create_instance = catalog_plan.schemas.service_instance.create.schema.to_json
+        rescue => e
+          @logger.error("Error parsing service_instance create schema: #{e.message}")
+        end
+
```

Comment:
I'm concerned about these rescues as the intent is unclear. Do you expect one of the fields to be nil and rescue a nil pointer exception? Do you expect a JSON marshal error due to bad input? Is there error actually cosmetic or was there a misunderstanding that `@logger.error` would re-raise?

Could you rewrite this to be more explicit about which failure mode you expect?

Reply:
As far as I can tell, we're just protecting against the absence of create_instance, update_instance, and/or create_binding schema (e.g. if catalog_plan.schemas.service_instance is nil, etc.). We shouldn't actually expect any exceptions aside from one that can be avoided by a conditional and some `try`ing.

Reply:
Looks like this was updated in this commit: 804050c91ee05d425b4637d85c8d451732df87c6

Reply:
We updated this and did some refactoring in 1046fd554920502ef2286d00491358f8e3bfcf14 

We no longer use rescue as a control flow instead we use try. I hope this makes the code a bit clearer.